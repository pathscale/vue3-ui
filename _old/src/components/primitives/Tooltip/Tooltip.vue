<script>
import { computed } from 'vue'

export default {
  name: 'VTooltip',
  props: {
    active: {
      type: Boolean,
      default: true
    },
    type: {
      type: String,
      default: 'is-primary'
    },
    label: {
      type: String,
      required: true
    },
    position: {
      type: String,
      default: 'is-top'
    },
    always: Boolean,
    animated: Boolean,
    square: Boolean,
    dashed: Boolean,
    multilined: Boolean,
    size: {
      type: String,
      default: 'is-medium'
    },
    delay: {
      type: Number,
      default: 0
    }
  },
  setup(props) {
    const transition = computed(() => {
      return {
        'transition-delay': `${props.delay}ms`
      }
    })
    return { transition }
  }
}
</script>

<template>
  <span
    :data-label="label"
    :class="[
      type,
      position,
      size,
      {
        'v-tooltip': active,
        'is-square': square,
        'is-animated': animated,
        'is-always': always,
        'is-multiline': multilined,
        'is-dashed': dashed
      }
    ]"
    :style="transition">
    <slot />
  </span>
</template>
