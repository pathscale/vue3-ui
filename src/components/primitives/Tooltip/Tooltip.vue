<script setup lang="ts">
import { computed } from "vue";

const props = withDefaults(
  defineProps<{
    active?: boolean;
    type?:
      | "is-white"
      | "is-light"
      | "is-dark"
      | "is-black"
      | "is-text"
      | "is-primary"
      | "is-link"
      | "is-info"
      | "is-success"
      | "is-warning"
      | "is-danger";
    label: string; // required
    position?: "is-top" | "is-bottom" | "is-left" | "is-right";
    always?: boolean;
    animated?: boolean;
    square?: boolean;
    dashed?: boolean;
    multilined?: boolean;
    size?: "is-small" | "is-medium" | "is-large";
    delay?: number;
  }>(),
  {
    active: true,
    type: "is-primary",
    position: "is-top",
    size: "is-medium",
    delay: 0,
  },
);

const rootStyles = computed(() => {
  return {
    "transition-delay": `${props.delay}ms`,
  };
});
</script>

<template>
  <span
    :data-label="label"
    :class="[
      type,
      position,
      size,
      {
        'v-tooltip': active,
        'is-square': square,
        'is-animated': animated,
        'is-always': always,
        'is-multiline': multilined,
        'is-dashed': dashed
      }
    ]"
    :style="rootStyles">
    <slot />
  </span>
</template>
