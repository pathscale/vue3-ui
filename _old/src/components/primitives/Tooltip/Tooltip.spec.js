import { shallowMount } from '@vue/test-utils'
import Tooltip from './Tooltip.vue'

let wrapper

describe('Tooltip', () => {
  beforeEach(() => {
    wrapper = shallowMount(Tooltip)
  })

  it('is called', () => {
    expect(wrapper.exists()).toBeTruthy()
  })

  it('render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
