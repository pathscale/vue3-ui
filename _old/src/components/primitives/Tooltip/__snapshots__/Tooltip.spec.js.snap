// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip render correctly 1`] = `<span class="is-primary is-top is-medium v-tooltip" style="transition-delay: 0ms;"></span>`;
